### PR TITLE
[INFRA] drop testing on python 3.7 and install via setup.cfg

### PR DESCRIPTION
**Pull request by Remi-Gau**
Would require relaxing the requirements on tenserflow.
https://discuss.tensorflow.org/t/support-python-3-10/124

Could possibly consider removing python 3.7 or wait for its end of life:
https://endoflife.date/python

### CODE REVIEW

**Comment by CYHSM** [comment]
Yes good idea, I think we can safely remove the requirement for TensorFlow. Do you want to add this to the pull request? Also would be nice to have the python versions be in the same format, thanks.

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -9,19 +9,21 @@ on:
...
+        python-version: ["3.8", "3.9", "3.10"] 
         include:
           - os: ubuntu-latest
             path: ~/.cache/pip
           - os: macOS-10.15
             path: ~/Library/Caches/pip
           - os: windows-latest
             path: ~\AppData\Local\pip\Cache
+            
+    runs-on: ${{ matrix.os }}
```
Comment (Remi-Gau) [inline .github/workflows/main.yml:25]: trying to see if actually running the OS part of the matrix runs or not

**Comment by Remi-Gau** [comment]
@CYHSM 

Seems that CI actually fails on Windows.

If you want to keep things clearer I can revert the last changes on this PR and keep its focus narrow on 3.10 and then open an issue and a new PR to tackle the Windows issue.

**Comment by CYHSM** [comment]
Yeah sounds good to me, I guess the problem might be antspy.

**Comment by CYHSM** [comment]
Ok seems to be the antspy problem ("Running setup.py install for antspyx: still running..."), so until this is solved upstream we will just support up to python 3.9, the other errors came either from changing v2->v3 or from matrix os (I guess the latter).

**Comment by Remi-Gau** [comment]
> the other errors came either from changing v2->v3 or from matrix os (I guess the latter).

yes most likely the latter because the way it is set up now, all tests are only run on Linux

See this workflow that is supposed to run on MacOS but actually ran on Ubuntu
https://github.com/DeepMReye/DeepMReye/runs/7576038296?check_suite_focus=true#step:1:3

So I would suggest merging this and fixing the CI for OS in another PR after that.

**Comment by Remi-Gau** [comment]
> Ok seems to be the antspy problem ("Running setup.py install for antspyx: still running..."), so until this is solved upstream we will just support up to python 3.9, the other errors came either from changing v2->v3 or from matrix os (I guess the latter).

dropping running the tests oon 3.10 for now until antspyx gets its 💩 together. :-)

**Pull request merged**